### PR TITLE
Rotational: default values are now guesses + fix tests

### PR DESCRIPTION
**Pull request by ven-k**
- Defaults in rotational lib are `guess`es
- Passes du0 and u0 for tests (in Rotational, Blocks)
- Fixes the linear analysis tests and docs (related to DC Motor example)

## Checklist

- [x] Appropriate tests were added
- [x] Any code changes were done in a way that does not break public API
- [x] All documentation related to code changes were updated
- [x] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [x] Any new documentation only uses public API

### CODE REVIEW

File: test/Mechanical/rotational.jl [hunk 1]
```diff
@@ -8,215 +8,219 @@ using OrdinaryDiffEq: ReturnCode.Success
...
+        end
+    end
 
-    @named model = ODESystem(connections, t,
-        systems = [fixed, inertia1, inertia2, spring, damper])
-    sys = structural_simplify(model)
+    @mtkbuild sys = TwoInertia()
 
-    prob = ODEProblem(sys, Pair[], (0, 10.0))
+    prob = ODEProblem(sys, [D(D(sys.inertia2.phi)) => 0], (0, 10.0))
```
Comment (ven-k) [inline test/Mechanical/rotational.jl:28]: @YingboMa this and in following ODE and DAE problems, now we have to pass an initial value to `sys₊inertia2₊phi_tt`. However, while adding this lib this was discussed and after some changes in tearing (in MTK), this was no longer required.
Why is it required again?

**Comment by ven-k** [comment]
@ChrisRackauckas this is ready for review.

File: src/Blocks/continuous.jl [hunk 2]
```diff
@@ -12,11 +12,15 @@ Initial value of integrator state ``x`` can be set with `x`
...
+
+# Unknowns:
+
+  - `x`: Unknown of Integrator (with initial guess of 0.0)
 """
 @mtkmodel Integrator begin
     @extend u, y = siso = SISO()
     @variables begin
-        x(t) = 0.0, [description = "State of Integrator"]
+        x(t), [description = "Unknown of Integrator", guess = 0.0]
```
Comment (baggepinnen) [inline src/Blocks/continuous.jl:23]: This instance of "state" should not be translated to "unknown", state is indeed correct here

File: test/Blocks/test_analysis_points.jl [hunk 3]
```diff
@@ -165,7 +166,7 @@ function SystemModel(u = nothing; name = :model)
            connect(inertia2.flange_a, spring.flange_b, damper.flange_b)]
     if u !== nothing
         push!(eqs, connect(torque.tau, u.output))
-        return @named model = ODESystem(eqs, t;
+        return ODESystem(eqs, t;
```
Comment (ChrisRackauckas) [inline test/Blocks/test_analysis_points.jl:169]: then the name needs to be passed

File: src/Mechanical/Rotational/components.jl [hunk 4]
```diff
@@ -167,7 +167,7 @@ Linear 1D rotational spring and damper
     @parameters begin
         d, [description = "Damping constant"]
         c, [description = "Spring constant"]
-        phi_rel0 = 0.0, [description = "Unstretched spring angle"]
+        phi_rel0, [description = "Unstretched spring angle", guess = 0.0]
```
Comment (ven-k) [inline src/Mechanical/Rotational/components.jl:170]: This is an oversight; will revert.

File: src/Blocks/continuous.jl (hunk 2 above)
Comment (ven-k) [reply src/Blocks/continuous.jl]: Resolved here: https://github.com/SciML/ModelingToolkitStandardLibrary.jl/pull/274/commits/eae592ba86e9de11fa575f005d3c13b526a80c99#diff-2c628e949fda2edfd7af3ee31fbf986535a6b1e959d92d03be9ca070a4b633aeR23

File: src/Mechanical/Rotational/components.jl (hunk 4 above)
Comment (ven-k) [reply src/Mechanical/Rotational/components.jl]: Resolved here: https://github.com/SciML/ModelingToolkitStandardLibrary.jl/pull/274/commits/eae592ba86e9de11fa575f005d3c13b526a80c99#diff-36a0d1920684771c0166099143e51f1cf58f1faae7d847cb151c0a0826c4348bR170

File: test/Blocks/test_analysis_points.jl (hunk 3 above)
Comment (ven-k) [reply test/Blocks/test_analysis_points.jl:169]: Resolved here: https://github.com/SciML/ModelingToolkitStandardLibrary.jl/pull/274/commits/eae592ba86e9de11fa575f005d3c13b526a80c99#diff-3603e44df7347f59dcb8354b6aa1af55ec51d86974ae7a5521ae7214cba54c27R177-R178

**Comment by ven-k** [comment]
Downgrade CI doesn't seem to be happy yet; but I will recheck up to which versions I can bump them to (as a follow up). I don't want to increase the compats to latest available versions; but keep them just above the minimum required ones.

**Comment by ven-k** [comment]
The failing Downgrade CI test is fixed by:
- https://github.com/SciML/ModelingToolkitStandardLibrary.jl/pull/276

**Comment by baggepinnen** [comment]
Considering how much of the user-facing documentation and tests had to be changed in this PR, I'd say this is a massively breaking change that requires a new major version?

File: src/Blocks/continuous.jl [hunk 5]
```diff
@@ -12,11 +12,15 @@ Initial value of integrator state ``x`` can be set with `x`
...
+
+# Unknowns:
+
+  - `x`: State of Integrator (with initial guess of 0.0)
 """
 @mtkmodel Integrator begin
     @extend u, y = siso = SISO()
     @variables begin
-        x(t) = 0.0, [description = "State of Integrator"]
+        x(t), [description = "State of Integrator", guess = 0.0]
```
Comment (baggepinnen) [inline src/Blocks/continuous.jl:23]: There is no guess needed here, this should be a default value.

File: src/Blocks/continuous.jl [hunk 6]
```diff
@@ -57,7 +66,7 @@ A smaller `T` leads to a more ideal approximation of the derivative.
 @mtkmodel Derivative begin
     @extend u, y = siso = SISO()
     @variables begin
-        x(t) = 0.0, [description = "Derivative-filter state"]
+        x(t), [description = "Derivative-filter state", guess = 0.0]
```
Comment (baggepinnen) [inline src/Blocks/continuous.jl:69]: Same here, there is no guessing involved, a default is much better.

**Comment by ven-k** [comment]
> I'd say this is a massively breaking change that requires a new major version?

This sounds good. Even the Units PR will be unblocked with this PR. So I'd say we can have that in too.
(And probably guesses in other sub-libraries as well - with dedicated PRs for each libraries)

> Was there a particular reasons these two blocks and only these were changed?

The idea is - incrementally change all the sub-libraries; tend to the ones with broken tests first.

The analysis-point tests were failing as the rotational components now have guesses instead of defaults - fixed with `op` - while there, I changed these two as they were involved.

I've reverted the changes to `Integrator` and `Derivative` blocks.

File: src/Blocks/continuous.jl (hunk 6 above)
Comment (ven-k) [reply src/Blocks/continuous.jl]: Resolved with: https://github.com/SciML/ModelingToolkitStandardLibrary.jl/pull/274/commits/3a7ccfbd43de07b1f1de3f81dc1b86e173146eac

File: src/Blocks/continuous.jl (hunk 5 above)
Comment (ven-k) [reply src/Blocks/continuous.jl]: Resolved with: https://github.com/SciML/ModelingToolkitStandardLibrary.jl/pull/274/commits/3a7ccfbd43de07b1f1de3f81dc1b86e173146eac

**Comment by baggepinnen** [comment]
This was released in a minor version but it should have been a major version bump 😔

**Comment by ChrisRackauckas** [comment]
It probably should have.

**Pull request merged**